### DISCUSSION TITLE
Music stops playing when entering dialogue

### DISCUSSION THREAD

**Discussion by nookstray** [dialogic-godot/dialogic#846] [Q&A] [answered]
I have music playing on my level, but when I enter dialogue, it stops. I’m guessing this is because I’m pausing the game to play the dialogue. Does anyone know a way around this? Maybe a way to only pause player movement and not audio? I would appreciate any help with this.

**Comment by Jowan-Spooner** [comment]
Yes this is caused by you pausing the game. Although I know that emilio did the same in his video, I would not suggest to using godots pause system for this, because it means, there is no way to actually pause your game anymore (for a pause menu of some sort).

I personally always use some kind of gamestate to tell the player and other things, that a dialog is running (or that other things are happening).

Here is a fictional example of how that could work:
```
# An Autoload Script called GameState.gd

enum States {None, Combat, Dialog}
var current_state = States .None
```

```
# when starting the dialog
func _on_NPC_start_dialog():
    if Dialogic.has_current_dialog_node():
         return
    var dialog = Dialogic.start('Overworld/Npc1/Greeting')
    dialog.connect('timeline_end', self, '_on_dialog_ended')
    add_child(dialog)
    GameState.current_state = GameState.States.Dialog

# once the dialog ended
func _on_dialog_ended(timeline_name):
    GameState.current_state = GameState.States.None

```

```
# how to avoid things when dialog is playing
func _physics_process(delta):
    # return in case there is currently a dialog playing
    if GameState.current_state == GameState.States.Dialog:
        return
    
    # if no dialog is playing do your movement
    if Input.is_action_pressed('crazy_movement'):
        do_crazy_movement()
```


*There are other options too. If you want to keep pausing the whole game you will need to set the pause_mode of the AudioPlayer to process. You can find that in the inspector at the very bottom.*

*If introducing a whole GameState doesn't make sense for your project you can just check agains Dialogic.has_current_dialog_node() which will return true whenever there is a dialog node running.*

**Comment by abbylian** [comment, marked as answer]
How is your music implemented? I resolved mine by putting AudioStreamPlayer node to Process (it's set to Inherit by default, if I remember correctly)

**Reply by nookstray** [reply]
Thank you so much! This resolved the issue.
